pkg/syncserver/sync_server.go

```diff
@@ -238,12 +245,30 @@ func (s *Server) serve(cxt context.Context) {
 	defer cancelFn()
 
 	logCxt := log.WithField("port", s.config.Port)
-	logCxt.Info("Opening listen socket")
-	l, err := net.ListenTCP("tcp", &net.TCPAddr{Port: s.config.ListenPort()})
+	pwd, _ := os.Getwd()
+	logCxt.WithField("pwd", pwd).Info("Opening listen socket")
+	var (
+		l   net.Listener
+		err error
+	)
+	if s.config.KeyFile != "" && s.config.CertFile != "" {
+		cert, err := tls.LoadX509KeyPair(s.config.CertFile, s.config.KeyFile)
+		if err != nil {
+			logCxt.WithError(err).Panic("Failed to load certificate and key")
+		}
+		tlsConfig := tls.Config{Certificates: []tls.Certificate{cert}}
+		tlsConfig.Rand = rand.Reader
+
+		laddr := fmt.Sprintf("0.0.0.0:%v", s.config.ListenPort())
+		l, err = tls.Listen("tcp", laddr, &tlsConfig)
```

Comment:
Do we actually validate the client's cert anywhere.  If we are provided with a CA cert, we should be validating client certs.